src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1943,7 +1943,11 @@ bool isRefEscape
                         parameters[argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex]] :
                         null;
 
-                    if (mixableArguments is not null && isMixableParameter(parameter))
+                    if (mixableArguments is not null
+                        && isMixableParameter(parameter)
+                        // assume any declaration variable is a valid mixing destination,
+                        // since we are going to infer a val-escape which makes it valid.
```

Comment:
```suggestion
                        // assume any expression variable is a valid mixing destination,
                        // since we will infer a legal val-escape for it (if it doesn't already have a narrower one).
```